GVFS/GVFS/RepairJobs/GitHeadRepairJob.cs

```diff
@@ -182,5 +103,17 @@ private bool CanBeRepaired(List<string> messages)
 
             return true;
         }
+
+        private string GetHeadRefSha()
+        {
+            string headRefFilePath = Path.Combine(this.Enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Head);
+
+            var contents = File.ReadAllText(headRefFilePath);
+            var sha = contents.Trim();
+
+            Debug.Assert(SHA1Util.IsValidShaFormat(sha), "Fix to HEAD ref should have written a valid SHA");

```

Comment:
Please don't use `Debug.Assert`. If it's a bug, it's a bug even in production, so just throw an exception.